lib/SILGen/SILGenGlobalVariable.cpp

```diff
@@ -110,12 +111,16 @@ SILGenFunction::emitGlobalVariableRef(SILLocation loc, VarDecl *var) {
   // Global variables can be accessed directly with global_addr.  Emit this
   // instruction into the prolog of the function so we can memoize/CSE it in
   // VarLocs.
-  auto entryBB = getFunction().getBlocks().begin();
-  SILGenBuilder prologueB(*this, entryBB, entryBB->begin());
-  prologueB.setTrackingList(B.getTrackingList());
+  auto entryBB = getFunction().begin();
+  std::unique_ptr<SILGenBuilder> prologueB;
+  if (entryBB->begin() == entryBB->end())
+    prologueB.reset(new SILGenBuilder(*this, entryBB));
+  else
+    prologueB.reset(new SILGenBuilder(*this, entryBB, entryBB->begin()));
```

Comment:
Can we push this condition into `SILGenBuilder::SILGenBuilder` instead? If not, it'd be preferable to use `Optional` instead of `unique_ptr` to avoid wanton heap allocation.